### PR TITLE
Flesh To Stone Statue Refactor

### PR DESCRIPTION
**Pull request by Incoming5643**
Removes the cruddy closet statues, people will now transition directly from human mobs to statue mobs, and simply can't do anything unless a staff of animation is used.

Depreciates the "whacking statues can destroy them" mechanic both because it's inconsistent with statue mobs and because it's almost completely unused. Due to this loss of utility the cooldown has been dropped 10 to 50 second. The Rank 5 cooldown likewise has dropped to 10 seconds from 20.

Statue mobs now can't attack their creators on a code level. Because people can't read.

### CODE REVIEW

File: code/game/gamemodes/wizard/spellbook.dm [hunk 1]
```diff
@@ -760,12 +760,12 @@
...
 	desc = "This book has a dedication to mimes everywhere inside the front cover."
 
-/obj/item/weapon/spellbook/oneuse/forcewall/recoil(mob/user)
+/*/obj/item/weapon/spellbook/oneuse/forcewall/recoil(mob/user)
 	..()
 	user <<"<span class='warning'>You suddenly feel very solid!</span>"
 	var/obj/structure/closet/statue/S = new /obj/structure/closet/statue(user.loc, user)
 	S.timer = 30
-	user.drop_item()
+	user.drop_item()*/
```
Comment (Incoming5643) [inline code/game/gamemodes/wizard/spellbook.dm]: Before you point it out, yes I forgot to do this bit, will fix.

**Comment by KorPhaeron** [comment]
Statue mobs attacking their owner would probably be more fair honestly, they're very powerful as is.

File: code/game/gamemodes/wizard/godhand.dm [hunk 2]
```diff
@@ -71,7 +71,38 @@
 	if(user.lying || user.handcuffed)
 		user << "<span class='warning'>You can't reach out!</span>"
 		return
-	var/mob/M = target
-	M.Stun(2)
-	new /obj/structure/closet/statue(M.loc, M)
-	..()
\ No newline at end of file
+	var/mob/living/L = target
+	if(!(ishuman(L) || ismonkey(L) || iscorgi(L)))
```
Comment (RemieRichards) [inline code/game/gamemodes/wizard/godhand.dm]: there's code below for monkey and corgi statues, yet you deny them here?

File: code/modules/mob/living/simple_animal/hostile/statue.dm [hunk 3]
```diff
@@ -45,12 +45,21 @@
 
 	var/cannot_be_seen = 1
 	var/mob/living/creator = null
+	var/active = 1
+	var/timer = -1
+
+	var/mob/living/victim = null
+	var/intialTox = 0 	//these are here to keep the internal mob from taking damage from things that logically wouldn't affect a rock
+	var/intialFire = 0	//it's a little sloppy I know but it was this or the GODMODE flag. Lesser of two evils.
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: Uh No, this is not the lesser of two evils, not in the slightest.

File: code/modules/mob/living/simple_animal/hostile/statue.dm [hunk 4]
```diff
@@ -83,10 +111,14 @@
...
 
 /mob/living/simple_animal/hostile/statue/AttackingTarget()
+	if(!active)
+		return
 	if(can_be_seen(get_turf(loc)))
 		if(client)
 			src << "<span class='warning'>You cannot attack, there are eyes on you!</span>"
 			return
+	if(target == creator) //Player statues aren't supposed to do this, and yet...
+		src << "<span class='warning'>You cannot attack your creator!</span>"
```
Comment (Supermichael777) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: shouldent this return?

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 3 above)
Comment (Incoming5643) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: Transplanted old code comments

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: still, fix their mess!

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 4 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: no need, the rest of attack code (```...()```) doesn't run unless this if fails.

**Comment by Firecage** [comment]
...Doesn't this mean that if you use flesh of stone of someone....you can't kill them anymore due to the god-like health? Since it was fun to turn someone into a statue, then destroy the statue to kill them.

File: code/game/gamemodes/wizard/spellbook.dm (hunk 1 above)
Comment (duncathan) [inline code/game/gamemodes/wizard/spellbook.dm]: Did you forget to do this bit?

**Comment by Zilenan91** [comment]
wait wait wait, does this mean that if you flesh to stone somebody, they're stuck as a statue forever? It switches their mob now, so...

**Comment by Incoming5643** [comment]
the timer is still there

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 3 above)
Comment (octareenroon91) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: Does it take at least two ticks for death to occur - one to take damage, one to die?
I'm seeing the possibility of accumulating radiation, hallucinate, maybe EMPs.

**Comment by Shadowlight213** [comment]
@incoming5643 pretty this fixes at least one issue.

**Comment by Incoming5643** [comment]
thought I was ready to finish this fight, but apparently not yet. Soon though.

**Pull request closed without merging**